### PR TITLE
FIX - streamline _build_train_config updates

### LINKED ISSUE #94: [BUG] [Pipeline] - streamline _build_train_config updates
Options to loss functions added but logic leftover from previous implementation which did not give options was left in, cleaning up config function and also adding ability to pass extra arguments to the loss function (such as `gamma` value, for which there is a 'catch' in `get_loss_function` but no ability to actually pass a value to it when running the training loop).

### PR DESCRIPTION
**Pull request by Mars-Martiny**
Fixes #94 
Cleaned up and reordered arguments for clarity. 
Values in config dictionary adjusted.

### CODE REVIEW

**Comment by AndreHrs** [comment]
Looks good, but flagging this to myself that #85 may have some issues due to change in training loop call.

Pasted Maryam's message here for xontext:
just be careful that when calling the main training function there is no longer a `criterion` argument (redundant) and there is now a criterion_args argument that expects a dict with any additional hyperparameters to for the loss function (focal expects 'gamma' and 'reduction')

**Review by AndreHrs — APPROVED** [review]

**Review by aashutosh-chapagain — COMMENTED** [review]
run_saving.py

```
self.history = {
    "training": {
        ...
        "epoch_time_sec": [],   # 👈 NESTED HERE
    },
    ...
}
```

But in train_loop.py, the code is still doing:

```
run_saver.history["epoch_time_sec"].append(...)
```

Now it should be:

```
run_saver.history["training"]["epoch_time_sec"]
```

The PR clearly moved all training-related metrics under "training"
epoch_time_sec is part of training history
So it belongs inside "training"

File: modules/training_loop/config.py
Comment (aashutosh-chapagain) [inline modules/training_loop/config.py:1]: need to fix datetime import
add:
```
from datetime import datetime
```
